.github/workflows/test-localnet-tests.yml

```diff
@@ -0,0 +1,55 @@
+name: Test localnet-dependent tests
+
+on:
+  pull_request:
+    branches: [main, feat/*]
+  workflow_dispatch:
+
+env:
+ BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+
+jobs:
+  localnet:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.11]
+
+    steps:
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install -r requirements.txt
+          pip3 install -r ./requirements-dev.txt --upgrade
+
+      - name: Set up MultiversX localnet
+        run: |
+          mkdir -p ~/multiversx-sdk
+          export PYTHONPATH=.
+          python3 -m multiversx_sdk_cli.cli localnet prerequisites --configfile=./multiversx_sdk_cli/tests/testdata/localnet_with_resolution_remote.toml
+          python3 -m multiversx_sdk_cli.cli localnet build --configfile=./multiversx_sdk_cli/tests/testdata/localnet_with_resolution_remote.toml
+         
+          # "Go" and artifacts from "GOPATH/pkg/mod" are not needed anymore.
+          sudo rm -rf ~/multiversx-sdk/golang
+          
+          python3 -m multiversx_sdk_cli.cli localnet clean --configfile=./multiversx_sdk_cli/tests/testdata/localnet_with_resolution_remote.toml
+          python3 -m multiversx_sdk_cli.cli localnet config --configfile=./multiversx_sdk_cli/tests/testdata/localnet_with_resolution_remote.toml
+          nohup python3 -m multiversx_sdk_cli.cli localnet start --configfile=./multiversx_sdk_cli/tests/testdata/localnet_with_resolution_remote.toml > localnet.log 2>&1 & echo $! > localnet.pid
+          sleep 120
+
+      - name: Test localnet dependent tests
+        run: |
+           pytest -m require_localnet .
+          
```

Comment:
Maybe we should stop the localnet as well?

https://github.com/multiversx/mx-sdk-js-core/blob/main/.github/workflows/test-localnet.yml#L61

Reply:
the runner dies immediately after the test if finished from what I know. I don't think there is a need to teardown